### PR TITLE
Add policy functionality

### PR DESCRIPTION
**Pull request by sai-c**
Added the policy functionality to rhub-api

### CODE REVIEW

File: src/openapi/openapi.yml [hunk 1]
```diff
@@ -439,9 +478,22 @@ components:
       required: true
       schema:
         $ref: '#/components/schemas/ID'
+    policy_search_term:
+      name: search_term
+      in: path
+      description: ID of the tower
```
Comment (brhatiga) [inline src/openapi/openapi.yml:484]: This might have been copied from something else?

**Review by brhatiga — COMMENTED** [review]
This is looking good!  Requested a review from @fpob

File: src/openapi/openapi.yml (hunk 1 above)
Comment (sai-c) [reply src/openapi/openapi.yml]: Thanks, Brian! I've just removed that specific parameter as it's no longer needed after the change to the search functionality.

**Review by fpob — CHANGES_REQUESTED** [review]
`tox -e flake8,yamllint`, there is a lot of linting errors

**Comment by fpob** [comment]
And could you sqash "Removed unneeded parameter definition" commits?

File: src/rhub/api/policies.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    new = {}
+    new["constraint"] = {}
+    for key, value in body.items():
+        if "constraint_" in key:
+            new["constraint"][key[11:]] = value
+        else:
+            new[key] = value
+    return new
+
+def check_token(func):
```
[the remark this answers is not in the document]
Comment (sai-c) [reply src/rhub/api/policies.py:66]: When I remove the token checking function, even though oauth2 is specified in openapi.yml, it seems to return data as long as the auth token field is not blank in the request. If I enter the oauth token as, "test", for example, the request (GET /policies) still returns data. If it is left blank, then "No authorization token provided" is returned. Please let me know what you think about this. Thanks.

File: src/rhub/api/policies.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    return inner
+
+
+def check_access(func):
+    """
+    Check whether user is in policy owners group
+    """
+    def inner(*args, **kwargs):
+        keycloak = get_keycloak()
+        status = kwargs.pop('token_info')['active']
```
[the remark this answers is not in the document]
Comment (sai-c) [reply src/rhub/api/policies.py:87]: See below comment.

File: src/rhub/api/policies.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fpob) [reply src/rhub/api/policies.py:66]: Then it is bug in `decode_token`.

File: src/rhub/api/policies.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        db.session.commit()
+        updated_record_row = model.Policy.query.filter(model.Policy.id == id).one()
+        updated_record_dict = row2dict(updated_record_row)
+    except exc.NoResultFound:
+        return problem(404, 'Not Found', 'Record Does Not Exist')
+    return denormalize(updated_record_dict)
+
+
+@check_access
+def delete_policy(user, id):
```
Comment (fpob) [inline src/rhub/api/policies.py:151]: `policy-...-owner` group in Keycloak can be also deleted. Group for a policy that doesn't exist will be useless I guess.

**Comment by sai-c** [comment]
Does this seem okay for merging?

**Review by fpob — APPROVED** [review]

**Pull request merged**